doc/manuals/contribution_guidelines.md

```diff
@@ -672,3 +684,18 @@ If you plan to use some of them, please consult before with the core developers.
   making the code error-prone due to subtle interactions. It gets worse in a multithreaded environment.
   * The flavor of this project is biased to C clearly. Many plain C libraries are used and mixing exceptions and
   traditional if-checking method should be avoided as much as possible.
+* Object passing. In order to avoid inefficient object copies when calling functions, the following criteria apply:
+  * If the function doesn't modify the object (i.e. "read only"), then use const references, e.g. `const BigFish& bf`
+  * If the function modifies the object (i.e. "read/write" or "write"), then use a pointer type, e.g: `BigFish* bf`. Note that in this case we could also use a reference, but we prefer pointers to make more explicit that the object is to be modified by the function.
+
+```
+void myFunction(const& BigFish in, BigFish* out)
+```
+
+* Strings return. In order to avoid inefficient object copies when returning strings, the  `const std::string&` return type is preferred. In the case that pattern cannot be used (e.g. when literal string such as "black", "red", etc. are used in the function) then `const char*` should be used as return type.
```

Comment:
If you find a better way of explaining in which cases `const std::string&` could be a problem and `const char*` has to be used, please go ahead.